Public/Src/Tools/SandboxedProcessExecutor/Executor.cs

```diff
@@ -0,0 +1,322 @@
+            info.StandardOutputObserver = m_outputErrorObserver.ObserveOutput;
+            info.StandardErrorObserver = m_outputErrorObserver.ObserveError;
+
+            return true;
+        }
+
+        private async Task<(ExitCode, SandboxedProcessResult)> ExecuteAsync(SandboxedProcessInfo info)
+        {
+            try
+            {
+                using (Stream standardInputStream = TryOpenStandardInputStream(info, out bool succeedInOpeningStdIn))
+                {
+                    if (!succeedInOpeningStdIn)
+                    {
+                        return (ExitCode.FailedSandboxPreparation, null);
+                    }
+
+                    using (StreamReader standardInputReader = standardInputStream == null ? null : new StreamReader(standardInputStream, CharUtilities.Utf8NoBomNoThrow))
+                    {
+                        info.StandardInputReader = standardInputReader;
+
+                        ISandboxedProcess process = await StartProcessAsync(info);
+
+                        if (process == null)
+                        {
+                            return (ExitCode.FailedStartProcess, null);
+                        }
+
+                        SandboxedProcessResult result = await process.GetResultAsync();
+
+                        // Patch result.
+                        result.WarningCount = m_outputErrorObserver.WarningCount;
+                        result.LastMessageCount = process.GetLastMessageCount();
+                        result.DetoursMaxHeapSize = process.GetDetoursMaxHeapSize();
+                        result.MessageCountSemaphoreCreated = info.FileAccessManifest.MessageCountSemaphore != null;
+
+                        return (ExitCode.Success, result);
+                    }
+                }
+            }
+            finally
+            {
+                info.FileAccessManifest.UnsetMessageCountSemaphore();
+            }
+        }
+
+        private async Task<ISandboxedProcess> StartProcessAsync(SandboxedProcessInfo info)
+        {
+            ISandboxedProcess process = null;
+            bool shouldRelaunchProcess = true;
+            int processRelaunchCount = 0;
+
+            while (shouldRelaunchProcess)
+            {
+                try
+                {
+                    shouldRelaunchProcess = false;
+                    process = await SandboxedProcessFactory.StartAsync(info, forceSandboxing: false);
+                }
+                catch (BuildXLException ex)
+                {
+                    if (ex.LogEventErrorCode == NativeIOConstants.ErrorFileNotFound)
+                    {
+                        ConsoleError($"Failed to start process: '{info.FileName}' not found");
+                        return null;
+                    }
+                    else if (ex.LogEventErrorCode == NativeIOConstants.ErrorPartialCopy && (processRelaunchCount < ProcessRelauchCountMax))
+                    {
+                        ++processRelaunchCount;

```

Comment:
>processRelaunchCount [](start = 26, length = 20)

Please log a message that we are going to launch the process again which should include the nr of times we will retry after this